### PR TITLE
RS+RT/YJ/Rule 22-16

### PR DESCRIPTION
**Pull request by yunjoonjung-PNNL**
I'm not so sure whether I developed [the temperature addition part](https://github.com/pnnl/ruleset-checking-tool/blob/develop/docs/section22/Rule22-16.md#:~:text=temperature%3A%20if%20supply_temperature_b%20%3D-,approach_b%20%2B%20wbt_b,-%3A%20PASS) in the ```rule_check``` method in the best way.

The problem was when the ```approach``` and ```wet-bulb``` temperatures were added, there was an error message saying "_{OffsetUnitCalculusError} Ambiguous operation with offset unit (degree_Celsius, degree_Celsius)._"  This is why I implemented it in a way that I converted the temperatures' magnitude, deg C, and then Kelvin. I'd appreciate it if you could take a look at this part more carefully. Thank you!

### CODE REVIEW

**Review by charliepnnl — APPROVED** [review]
Reviewed section22rule16.py

Approved with a few comments that should be addressed.

File: rct229/rules/section22/section22rule16.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            super(Section22Rule16.ChillerHeatRejectionRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, False),
+                required_fields={
+                    "$": ["design_wetbulb_temperature", "approach", "loop"],
+                },
+            )
+
+        def get_calc_vals(self, context, data=None):
+            heat_rejection_b = context.baseline
+            wbt_b = heat_rejection_b["design_wetbulb_temperature"]
```
[the remark this answers is not in the document]
Comment (yunjoonjung-PNNL) [reply rct229/rules/section22/section22rule16.py:84]: Fixed.

File: rct229/rules/section22/section22rule16.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+            <= REQUIRED_HIGH_DESIGN_WETBULB_TEMP.to(ureg.kelvin)
+            for heat_rejection_wetbulb_temp_b in find_all(
+                "heat_rejections[*].design_wetbulb_temperature", rmi_b
+            )
+        )
+
+    def create_data(self, context, data):
+        rmi_b = context.baseline
+        supply_temperature_dict_b = {
+            heat_rejection_loop: find_exactly_one_with_field_value(
```
Comment (weilixu) [inline rct229/rules/section22/section22rule16.py:66]: As discussed yesterday, this portion needs to be updated.

File: rct229/rules/section22/section22rule16.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+            }
+
+        def rule_check(self, context, calc_vals=None, data=None):
+            wbt_b = calc_vals["wbt_b"]
+            approach_b = calc_vals["approach_b"]
+            loop_b = calc_vals["loop_b"]
+            supply_temperature_b = data["supply_temperature_dict_b"][loop_b]
+            return std_equal(
+                supply_temperature_b.to(ureg.kelvin),
+                ((wbt_b.m + approach_b.m) * ureg("degC")).to(ureg.kelvin),
```
[the remark this answers is not in the document]
Comment (charliepnnl) [reply rct229/rules/section22/section22rule16.py]: Oh, wait.  Yes, it makes no sense to add two measured temperatures together. One of these must actually be a temperature difference.  I thought the schema used kelvin for all temperature differences and Celsius for all measured temperatures.  It think there might be a problem with the schema.

@weilixu @yunjoonjung-PNNL   is approach a temperature difference?  If so I think there is a problem with the schema that we should bring up to Jason.

[the remark this answers is not in the document]
Comment (weilixu) [reply rct229/rules/section22/section22rule16.py]: #691

**Comment by weilixu** [comment]
@yunjoonjung-PNNL The schema is updated with the correct unit (still 0.0.18). You can use this schema to complete the task

File: rct229/rules/section22/section22rule16.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
+from rct229.rule_engine.rule_base import RuleDefinitionBase
+from rct229.rule_engine.rule_list_indexed_base import RuleDefinitionListIndexedBase
+from rct229.rule_engine.user_baseline_proposed_vals import UserBaselineProposedVals
+from rct229.schema.config import ureg
+from rct229.utils.assertions import getattr_
+from rct229.utils.jsonpath_utils import find_all, find_exactly_one_with_field_value
+from rct229.utils.pint_utils import CalcQ
+from rct229.utils.std_comparisons import std_equal
+
+APPLICABLE_SYS_TYPES = [
```
Comment (weilixu) [inline rct229/rules/section22/section22rule16.py:10]: The `get_baseline_system_types` function is merged. Please update the `APPLICABLE_SYS_TYPES` list and `is_applicable` function in this rule definition

File: rct229/rules/section22/section22rule16.py (hunk 2 above)
Comment (yunjoonjung-PNNL) [reply rct229/rules/section22/section22rule16.py:64]: Fixed. Thank you Weili!

File: rct229/rules/section22/section22rule16.py (hunk 4 above)
Comment (yunjoonjung-PNNL) [reply rct229/rules/section22/section22rule16.py:10]: Fixed.

**Review by weilixu — APPROVED** [review]

**Pull request merged**